compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -1618,6 +1618,12 @@ pub enum StatementKind<'tcx> {
     /// parts of this may do type specific things that are more complicated than simply copying
     /// bytes.
     ///
+    /// For `Rvalue::Use`, `Rvalue::Aggregate`, and `Rvalue::Repeat`, the destination `Place` may
+    /// not overlap with any memory computed as a part of the `Rvalue`. Relatedly, the evaluation
+    /// order is `compute lhs place -> compute rhs value -> store value in place`. The plan is for
+    /// the rule about the evaluation order to justify the rule about the non-overlappingness via
+    /// the aliasing model.
```

Comment:
Oh wait, having the eval order depend on the rvalue is *deliberate*? That thought didn't even occur to me when reading the `step.rs` changes.

I think this is a very bad idea. It makes justifying rvalue transformations a total disaster.